stdlib/set.mli

```diff
@@ -107,131 +104,53 @@ module type S =
-    (** [iter f s] applies [f] in turn to all elements of [s].
-       The elements of [s] are presented to [f] in increasing order
-       with respect to the ordering over the type of the elements. *)
-
-    val map: (elt -> elt) -> t -> t
-    (** [map f s] is the set whose elements are [f a0],[f a1]... [f
-        aN], where [a0],[a1]...[aN] are the elements of [s].
-
-       The elements are passed to [f] in increasing order
-       with respect to the ordering over the type of the elements.
-
-       If no element of [s] is changed by [f], [s] is returned
-       unchanged. (If each output of [f] is physically equal to its
-       input, the returned set is physically equal to [s].)
-       @since 4.04.0 *)
-
-    val fold: (elt -> 'a -> 'a) -> t -> 'a -> 'a
-    (** [fold f s init] computes [(f xN ... (f x2 (f x1 init))...)],
-       where [x1 ... xN] are the elements of [s], in increasing order. *)
-
-    val for_all: (elt -> bool) -> t -> bool
-    (** [for_all f s] checks if all elements of the set
-       satisfy the predicate [f]. *)
-
-    val exists: (elt -> bool) -> t -> bool
-    (** [exists f s] checks if at least one element of
-       the set satisfies the predicate [f]. *)
-
-    val filter: (elt -> bool) -> t -> t
-    (** [filter f s] returns the set of all elements in [s]
-       that satisfy predicate [f]. If [f] satisfies every element in [s],
-       [s] is returned unchanged (the result of the function is then
-       physically equal to [s]).
-       @before 4.03 Physical equality was not ensured.*)
-
-    val filter_map: (elt -> elt option) -> t -> t
-    (** [filter_map f s] returns the set of all [v] such that
-        [f x = Some v] for some element [x] of [s].
-
-       For example,
-       {[filter_map (fun n -> if n mod 2 = 0 then Some (n / 2) else None) s]}
-       is the set of halves of the even elements of [s].
-
-       If no element of [s] is changed or dropped by [f] (if
-       [f x = Some x] for each element [x]), then
-       [s] is returned unchanged: the result of the function
-       is then physically equal to [s].
-
-       @since 4.11.0
-     *)
-
-    val partition: (elt -> bool) -> t -> t * t
-    (** [partition f s] returns a pair of sets [(s1, s2)], where
-       [s1] is the set of all the elements of [s] that satisfy the
-       predicate [f], and [s2] is the set of all the elements of
-       [s] that do not satisfy [f]. *)
+        that are not in [s2]. *)
 
     val cardinal: t -> int
     (** Return the number of elements of a set. *)
 
+    (** {1:elements Elements} *)
+
     val elements: t -> elt list
     (** Return the list of all elements of the given set.
-       The returned list is sorted in increasing order with respect
-       to the ordering [Ord.compare], where [Ord] is the argument
-       given to {!Set.Make}. *)
+        The returned list is sorted in increasing order with respect
+        to the ordering [Ord.compare], where [Ord] is the argument
```

Comment:
Is odoc smart enough to understand `{!Ord.compare}` these days? That might be a useful cross-reference here.

Reply:
Likely but it's not the goal of this PR to improve the docstrings of the module.